frontend/src/pages/modelServing/screens/metrics/bias/BiasChart.tsx

```diff
@@ -19,28 +19,33 @@ const BiasChart: React.FC<BiasChartProps> = ({ biasMetricConfig }) => {
     BIAS_CHART_CONFIGS[metricType];
 
   const metric = React.useMemo(() => {
-    const metricData = data[modelMetricKey].data as PrometheusQueryRangeResponseDataResult[];
-
-    const values = metricData.find((x) => x.metric.request === id)?.values || [];
-
-    return {
-      ...data[modelMetricKey],
-      data: values,
-    };
+    const metricData = data[modelMetricKey].data;
+    if (isPrometheusQueryRangeResponseDataResultArr(metricData)) {
```

Comment:
Look into fixing the type from the context so that the keys always gives back the matching typed value.